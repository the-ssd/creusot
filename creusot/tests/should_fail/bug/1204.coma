module M_1204__evil [#"1204.rs" 8 0 8 22]
  let%span s12040 = "1204.rs" 6 11 6 16
  let%span s12041 = "1204.rs" 7 10 7 11
  let%span s12042 = "1204.rs" 9 4 9 16
  
  use prelude.prelude.Int
  
  constant x  : int
  
  function evil'0 [#"1204.rs" 8 0 8 22] (x : int) : int
  
  goal vc_evil'0 : ([%#s12040] false)
   -> ([@expl:evil requires] [%#s12040] false) /\ 0 <= ([%#s12041] x) /\ ([%#s12041] - x) < ([%#s12041] x)
end
module M_1204__wrong [#"1204.rs" 14 0 14 10]
  let%span s12040 = "1204.rs" 13 10 13 15
  let%span s12041 = "1204.rs" 6 11 6 16
  let%span s12042 = "1204.rs" 7 10 7 11
  let%span s12043 = "1204.rs" 15 4 15 43
  let%span s12044 = "1204.rs" 9 4 9 16
  
  use prelude.prelude.Int
  
  function evil'0 [#"1204.rs" 8 0 8 22] (x : int) : int
  
  axiom evil'0_def : forall x : int . ([%#s12041] false)  -> evil'0 x = ([%#s12044] evil'0 (- x) + 1)
  
  constant _1  : ()
  
  function wrong'0 [#"1204.rs" 14 0 14 10] (_1 : ()) : ()
  
  goal vc_wrong'0 : ([@expl:evil requires] [%#s12041] false)
  /\ ([@expl:evil requires] [%#s12041] false)
  /\ evil'0 1 = evil'0 (- 1) + 1
  && (let _ = () in let _ = () in ([@expl:evil requires] [%#s12041] false)
  /\ ([@expl:evil requires] [%#s12041] false) /\ evil'0 (- 1) = evil'0 1 + 1 && (let _ = () in [%#s12040] false))
end
