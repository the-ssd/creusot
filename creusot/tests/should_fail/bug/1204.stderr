warning: function cannot return without recursing
 --> 1204.rs:8:1
  |
8 | fn evil(x: Int) -> Int {
  | ^^^^^^^^^^^^^^^^^^^^^^ cannot return without recursing
9 |     evil(-x) + 1
  |     -------- recursive call site
  |
  = help: a `loop` may express intention better if this is on purpose
  = note: `#[warn(unconditional_recursion)]` on by default

warning: function `evil` is never used
 --> 1204.rs:8:4
  |
8 | fn evil(x: Int) -> Int {
  |    ^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: function `wrong` is never used
  --> 1204.rs:14:4
   |
14 | fn wrong() {
   |    ^^^^^

warning: 3 warnings emitted

